helix-core/src/syntax.rs

```diff
@@ -1144,6 +1144,8 @@ pub enum HighlightEvent {
     Source { start: usize, end: usize },
     HighlightStart(Highlight),
     HighlightEnd,
+    SelectionStart,
+    SelectionEnd,
```

Comment:
Sorry it took a while! I wonder if we could avoid additional event types and instead just track this in the render loop. When `HighlightStart(selection scope)` is emitted we'd toggle a local bool to true, then false when it's popped off the scope stack

Reply:
Done.  I hope I've understood your suggestion properly

Reply:
I think archseer is saying that you can track this without changing the `helix_core::syntax` by looking up the scope for the highlight event and seeing if it's a selection scope:

```rs
HighlightEvent::HighlightStart(span) => {
    if theme
        .scopes()
        .get(span.0)
        .map(|scope| scope.starts_with("ui.selection"))
        .unwrap_default()
    {
        selecting = true;
    }
    spans.push(span);
}
// and for HighlightEnd, check that the popped scope starts with "ui.selection"
```

that snippet ☝️ isn't very performant though - we should calculate which indices in `theme.scopes()` correspond to selection highlights outside of the `for`

Reply:
I see, thanks for clarifying.  I think I've done it in the latest revision